web/modules/custom/dpl_fees/src/Form/FeesListSettingsForm.php

```diff
@@ -0,0 +1,153 @@
+    return [
+      $this->configService->getConfigKey(),
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId(): string {
+    return 'fee_list_settings_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    $config = $this->configService->getConfig();
+
+    $form['settings'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Basic settings'),
+      '#tree' => FALSE,
+    ];
+
+    $form['settings']['fees_and_replacement_costs_url'] = [
+      '#type' => 'url',
+      '#title' => $this->t('Fees and Replacement costs URL'),
+      '#description' => $this->t('File or URL containing the fees and replacement costs'),
+      '#default_value' => $config['feesAndReplacementCostsUrl'] ?? '',
+    ];
+
+    $form['settings']['available_payment_types_url'] = [
+      '#type' => 'url',
+      '#title' => $this->t('Available payment types url'),
+      '#default_value' => $config['availablePaymentTypesUrl'] ?? '',
+    ];
+
+    $form['settings']['terms_of_trade_text'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Terms of trade text'),
+      '#description' => $this->t('Terms of trade text'),
+      '#default_value' => $config['termsOfTradeText'] ?? '',
+    ];
+
+    $form['settings']['terms_of_trade_url'] = [
+      '#type' => 'url',
+      '#title' => $this->t('Terms of trade redirect url'),
+      '#default_value' => $config['termsOfTradeUrl'] ?? '',
+    ];
+
+    $form['settings']['page_size_mobile'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Page size mobile', [], ['context' => 'Dashboard (settings)']),
+      '#default_value' => $config['pageSizeMobile'] ?? 25,
+      '#min' => 0,
+      '#step' => 1,
+    ];
+
+    $form['settings']['page_size_desktop'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Page size desktop', [], ['context' => 'Dashboard (settings)']),
+      '#default_value' => $config['pageSizeDesktop'] ?? 25,
+      '#min' => 0,
+      '#step' => 1,
+    ];
+
+    // @todo images to be done in future render.
+    $form['settings']['image'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Payment options image'),
+      '#description' => $this->t('Image containing the available payment options (300x35)'),
+      '#default_value' => $config['image'] ?? '',
+    ];
+
+    $form['settings']['payment_overview_url'] = [
+      '#type' => 'url',
+      '#title' => $this->t('Payment overview url'),
+      '#default_value' => $config['paymentOverviewUrl'] ?? '',
+    ];
+
+    $form['settings']['fee_list_body_text'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Intro text'),
+      '#description' => $this->t('Display an intro-text below the headline'),
+      '#default_value' => $config['feeListBodyText'] ?? $this->t('Fees and replacement costs are handled through the new system "Mit betalingsoverblik.'),
+    ];
```

Comment:
Just curious: No fields are required?

Reply:
No, I could make them, but they have not been very specific about this in the specification of the system. So we'll talk to them about this during a test meeting.